driver-async/src/main/com/mongodb/async/client/MongoClients.java

```diff
@@ -213,9 +215,10 @@ public static CodecRegistry getDefaultCodecRegistry() {
 
     private static final CodecRegistry DEFAULT_CODEC_REGISTRY =
             fromProviders(asList(new ValueCodecProvider(),
-                    new DocumentCodecProvider(),
                     new BsonValueCodecProvider(),
-                    new IterableCodecProvider(),
+                    new DBRefCodecProvider(),
+                    new DocumentCodecProvider(new DocumentToDBRefTransformer()),
```

Comment:
OK, let's bring them into line then.  It should be called out in the release notes.